indice_pollution/history/models/indice_uv.py

```diff
@@ -0,0 +1,130 @@
+from indice_pollution.extensions import db
+from indice_pollution.helpers import today
+from indice_pollution.history.models.commune import Commune
+from dataclasses import dataclass
+from datetime import datetime
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+import os, logging, csv
+
+
+from ftplib import FTP
+import io
+
+@dataclass
+class IndiceUv(db.Model):
+    __table_args__ = {"schema": "indice_schema"}
+
+    zone_id: int = db.Column(db.Integer, db.ForeignKey('indice_schema.zone.id'), primary_key=True)
+    date: datetime.date = db.Column(db.Date, primary_key=True, nullable=False)
+    uv_j0: int = db.Column(db.Integer)
+    uv_j1: int = db.Column(db.Integer)
+    uv_j2: int = db.Column(db.Integer)
+    uv_j3: int = db.Column(db.Integer)
+
+    @classmethod
+    def save_all(cls):
+
+        def get_ftp_content():
+            if not os.getenv('FS_BUCKET_URL') or not os.getenv('FS_BUCKET_USERNAME') or not os.getenv('FS_BUCKET_PASSWORD'):
+                return None
+            try:
+                ftp = FTP(os.getenv('FS_BUCKET_URL'))
+                ftp.login(os.getenv('FS_BUCKET_USERNAME'), os.getenv('FS_BUCKET_PASSWORD'))
+                bio = io.BytesIO()
+                def handle_binary(data):
+                    bio.write(data)
+                filename = f'{today().strftime("%Y%m%d")}'
+                ftp.retrbinary(f'RETR {filename}.csv', callback=handle_binary)
+                b = bio.getvalue()
+                decoded_content = b.decode('UTF-8')
+            except Exception as e:
+                logging.error(e)
+                return None
+            return decoded_content
+
+        decoded_content = get_ftp_content()
+        if decoded_content is None:
+            return
+        reader = csv.DictReader(
+            decoded_content.splitlines(),
+            delimiter=';'
+        )
+        indices = []
+
+        def format_insee(raw_insee: str):
+            if len(raw_insee) == 5 and raw_insee.isdigit():
+                return raw_insee
+            else:
+                return None
+        def format_uv(raw_uv: str): # '-999' or '' means no value
+            if raw_uv.isdigit():
+                uv = int(raw_uv)
+                if uv < 0:
+                    uv = None
+            else:
+                uv = None
+            return uv
+
+        # file header: insee;commune;date;UV_J0;UV_J1;UV_J2;UV_J3
+        for row in reader:
+            insee = format_insee(row['insee'])
+            oudated_communes = []
+            if insee and insee not in oudated_communes:
+                departement_code = f'{insee:0>2}'
+                if departement_code == '20': # Corse: 20 in file, 2A or 2B in db
+                    insee_2a = insee.replace('20', '2A')
```

Comment:
On ne veut remplacer que la première occurence
```suggestion
                    insee_2a = insee.replace('20', '2A', 1)
```